spec/abstractops.html

```diff
@@ -1098,14 +1099,25 @@ <h1>ISO 8601 grammar</h1>
       TimeFraction :
           Fraction
 
-      TimeZoneUTCOffset :
-          UTCOffset
+      UTCOffsetMinutePrecision :
+          TimeZoneUTCOffsetName[+Extended]
+          TimeZoneUTCOffsetName[~Extended]
+
+      UTCOffsetWithSubMinute[Extended] :
+          Sign Hour[+Padded] TimeSeparator[?Extended] MinuteSecond TimeSeparator[?Extended] MinuteSecond Fraction?
+
+      UTCOffsetNanosecondPrecision :
           UTCDesignator
+          UTCOffsetMinutePrecision
+          UTCOffsetWithSubMinute[+Extended]
+          UTCOffsetWithSubMinute[~Extended]
+
+      NormalizedUTCOffset :
+          ASCIISign Hour[+Padded] `:` MinuteSecond
 
       TimeZoneUTCOffsetName[Extended] :
           Sign Hour[+Padded]
           Sign Hour[+Padded] TimeSeparator[?Extended] MinuteSecond
-          Sign Hour[+Padded] TimeSeparator[?Extended] MinuteSecond TimeSeparator[?Extended] MinuteSecond Fraction?
```

Comment:
Refactor looks good, but after trying exactly the naming you recommended (where nano precision is just |UTCOffset|) and trying the current PR text's naming (where both the nano and the minute version have the precision in the name) is that I found it easier to reason about which call sites use which variant of the offset syntax when I didn't have to keep remembering which was the nano version and which was the minutes version.  

Also, if we did call one of them just plain |UTCOffset|, then I'd probably suggest it be the minutes version, because that's the one supported by external standards and also the one actually supported by current shipping implementations.  The nano version, being the new kid on the block, seems like it deserves a special name to let folks know that it's a standards deviation. 

What do you think?

Reply:
I guess I'd want to see a side-by-side comparison, but my gut reaction is that the name qualifiers matter less than the grammar structure.

Reply:
OK cool. I'll plan to use the structure you suggested but probably will use a different name than |UTCOffset|, for clarity.

Reply:
OK, refactor is pushed. Want to take a look and let me know what you think?